tests/conftest.py

```diff
@@ -304,12 +304,7 @@ def get_logger() -> GetLogger:
 
 
 @pytest.fixture()
-async def redis_client(docker_ip: str, redis_service: None) -> AsyncGenerator[AsyncRedis, None]:
-    client: AsyncRedis = AsyncRedis(host=docker_ip, port=6397)
-    yield client
-    try:
-        await client.aclose()  # type: ignore[attr-defined]
-    except RuntimeError:
-        pass
-    finally:
+async def redis_client(docker_ip: str, redis_service: None) -> AsyncRedis:
+    async with AsyncRedis(host=docker_ip, port=6397) as client:
         await client.flushall()
+        return client
```

Comment:
I don't understand why this works. The context manager closes the client immediately before returning so won't the client be closed for all the tests?

Reply:
now you mention it this is weird indeed,
https://github.com/litestar-org/litestar/pull/3117 is the change for the redis fixture and all tests pass fine, yielding the client and nothing works,

Reply:
given https://docs.python.org/3/reference/compound_stmts.html#async-with

we have then
```
    async with AsyncRedis(host=docker_ip, port=6397) as client:
        await client.flushall()
        return client
```
is in fact (I may have syntax errors here it's just to understand the flow)

```
hit_except = False

try:
    client = AsyncRedis.__aenter__()
    await client.flushall()
    return client
except:
    hit_except = True
    if not await aexit(manager, *sys.exc_info()):
        raise
finally:
    if not hit_except:
        await AsyncRedis.__aexit()
```